src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -2262,17 +2317,133 @@ export class StudyViewPageStore {
+    updateUserSettings(forceSave?:boolean) {
+        // forceSave => instantaneously save user settings independent of whether there are any changes 
+        // between current and previously saved settings
+        clearTimeout(this.updateSettingsTimeout);
+        //instantly save settings for any of these conditions
+        const timeout = forceSave || _.isEmpty(this.currentChartSettingsMap) ? 0 : 3000;
+
+        this.updateSettingsTimeout = setTimeout(() => {
+            if(forceSave || !_.isEqual(this.previousSettings, this.currentChartSettingsMap)){
+                this.previousSettings = this.currentChartSettingsMap;
+                if(!_.isEmpty(this.currentChartSettingsMap) && this.userLoggedIn) {
+                    sessionServiceClient.updateUserSettings({
+                        origin: toJS(this.studyIds),
+                        chartSettings: _.values(this.currentChartSettingsMap)
+                    });
+                }
+            }
+        }, timeout)
+    }
+
+    @computed get currentChartSettingsMap(){
+        let settings: { [chartId: string]: ChartUserSetting } = {};
+            this._chartVisibility.entries().forEach(([id, visible]) => {
+                if (visible) {
+                    settings[id] = {
+                        id,
+                        chartType: this.chartsType.get(id),
+                        patientAttribute: this.chartMetaSet[id].patientAttribute
+                    }
+
+                    const customChart = this._customChartMap.get(id);
+                    if (customChart) {
+                        settings[id].groups = customChart.groups;
+                        settings[id].name = this.chartMetaSet[id].displayName;
+                    }
+                }
+            });
+
+            this.currentGridLayout.forEach(layout => {
+                if (layout.i && settings[layout.i]) {
+                    settings[layout.i].layout = {
+                        x: layout.x,
+                        y: layout.y,
+                        w: layout.w,
+                        h: layout.h
+                    };
+                }
+            });
+        return settings;
+    }
+
+    readonly fetchUserSettings = remoteData<ChartUserSetting[]>({
+        invoke: async () => {
+            if (this.userLoggedIn && this.studyIds.length > 0) {
+                let userSettings = await sessionServiceClient.fetchUserSettings(toJS(this.studyIds));
+                return userSettings.chartSettings || [];
+            }
+            return [];
+        },
+        default: [],
+        onError:()=>{
+            // fail silently when an error occurs
+        }
+    });
+
+    @autobind
+    @action
+    loadUserSettings() {
+        if (!_.isEmpty(this.fetchUserSettings.result)) {
+            this.previousSettings = _.keyBy(this.fetchUserSettings.result, chartSetting => chartSetting.id)
+            this._chartVisibility.clear();
+            this.currentGridLayout = [];
```

Comment:
We should have a function to clear up the layout settings. I think it's necessary `currentFocusedChartByUser` and `currentFocusedChartByUserDimension` should be set back to original state as well.